src/interfaces/background_field_interface.jl

```diff
@@ -0,0 +1,198 @@
+####################
+# The abstract background field interface
+#
+# In this file, the abstract interfaces for different types of background fields
+# is defined. 
+####################
+"""
+Abstract base type for describing classical background fields.
+
+The only supported class of background fields yet, are the pulsed plane-wave field (see [`AbstractPulsedPlaneWaveField`](@ref) for details).
+"""
+abstract type AbstractBackgroundField end
+
+#####
+# pulsed plane-wave fields
+#####
+
+"""
+Abstract base type for pulsed plane-wave fields.
+
+!!! note "Pulsed field interface"
+
+    For every subtype, the following functions need to be implemented.
+
+    ```Julia
+
+    reference_momentum(::AbstractPulsedPlaneWaveField)
+    domain(::AbstractPulsedPlaneWaveField)
+    phase_duration(::AbstractPulsedPlaneWaveField)
+    envelope(::AbstractPulsedPlaneWaveField,x::Real)
```

Comment:
Is this list missing `amplitude()`? Since you use it in `generic_spectrum()` which is exported in `src/QEDfields.jl`.

Reply:
This is the list of interface functions to be implemented. For the function `amplitude`, a default implementation is given, which uses these interface functions. In this sense, `amplitude` and `generic_spectrum` have the same *status*: functions derived from the interface.